localstack/services/cloudformation/service_models.py

```diff
@@ -1394,9 +1394,12 @@ def cloudformation_type():
         return 'AWS::EC2::VPC'
 
     def fetch_state(self, stack_name, resources):
+        vpc_id = self.physical_resource_id
```

Comment:
I think we should generally discuss whether we want to go with attribute-based `fetch_state` implementations, or with `PhysicalResourceId` based `fetch_state` implementations.

Both have their pros and cons, but I would generally try to be more lenient when it comes to deploying stacks into environments where certain resources already exist.

In other words: The previous implementation allowed CF stacks to be deployed in a LocalStack instance where a VPC with a particular `CidrBlock` already exists. With the new implementation (after your changes), the CF stack creation will likely fail if such a VPC already exists (because `self.physical_resource_id` is None, and when CF tries to run `create_vpc(..)` it will fail because a VPC with this `CidrBlock` already exists).

From a usability point of view, I think we should rather be more lenient (less strict) and try use the **attribute-based `fetch_state` implementation** (whenever possible). I can, for example, anticipate issues where we may have different support levels for persistence - e.g., if certain resources are already deployed because they have persistence enabled, and then CF deployments would fail.

---
To **summarize** the points above: I think we should revert these changes and leave the original implementation, to avoid users running into weird issues in different use cases. Thanks